Fix types import/exports

skogsmaskin opened the pull request:
### Description

Fix type import/exports to work with vite.js

<!--
What changes are introduced?
Why are these changes introduced?
What issue(s) does this solve? (with link, if possible)
-->

### What to review

<!--
What steps should the reviewer take in order to review?
What parts/flows of the application/packages/tooling is affected?
Add a test script, if that makes sense.
-->

### Notes for release

<!--
A description of the change(s) that should be used in the release notes.
-->

bjoerge commented on packages/@sanity/base/src/components/previews/types.ts:
```diff
@@ -11,8 +11,8 @@ export type FieldName = string
 export type Selection = [Id, FieldName[]]
-export {PrepareViewOptions} from '@sanity/types'
-export {SortOrdering} from '@sanity/types'
+export type {PrepareViewOptions} from '@sanity/types'
+export type {SortOrdering} from '@sanity/types'
```
```suggestion
export type {PrepareViewOptions, SortOrdering} from '@sanity/types'
```

bjoerge commented on packages/@sanity/base/src/preview/types.ts:
```diff
@@ -1,7 +1,7 @@
 
-export {PrepareViewOptions} from '@sanity/types'
-export {SortOrdering} from '@sanity/types'
+export type {PrepareViewOptions} from '@sanity/types'
+export type {SortOrdering} from '@sanity/types'
```
```suggestion
export type {SortOrdering, PrepareViewOptions} from '@sanity/types'
```

bjoerge commented on packages/@sanity/base/src/_exports/_internal.ts:
```diff
@@ -11,7 +11,8 @@ export {observePathsDocumentPair as unstable_observePathsDocumentPair} from '../
 export {observeDocumentPairAvailability as unstable_observeDocumentPairAvailability} from '../preview/availability'
 
-export {DocumentAvailability, AvailabilityReason} from '../preview/types'
+export type {DocumentAvailability} from '../preview/types'
+export {AvailabilityReason} from '../preview/types'
```
```suggestion
export {type AvailabilityReason, DocumentAvailability} from '../preview/types'
```

skogsmaskin replied:
This one doesn't seem like a valid way to do the export.

bjoerge replied:
it is valid typescript, but maybe other parsers haven't caught up yet

skogsmaskin replied:
Must be `export type {AvailabilityReason, DocumentAvailability} from '../preview/types'`

bjoerge reviewed (approved):
Thanks! Added some suggestions, feel free to ignore as they are not really related to the problem you're fixing here 😄

skogsmaskin commented:
> Thanks! Added some suggestions, feel free to ignore as they are not really related to the problem you're fixing here smile

I have added your suggestions and squashed them into my commit.
